c_glib/arrow-glib/reader.cpp

```diff
@@ -1497,6 +1501,393 @@ garrow_csv_reader_read(GArrowCSVReader *reader,
   }
 }
 
+
+typedef struct GArrowJSONReadOptionsPrivate_ {
+  arrow::MemoryPool *pool;
+  arrow::json::ReadOptions read_options;
+  arrow::json::ParseOptions parse_options;
+} GArrowJSONReadOptionsPrivate;
+
+enum {
+  PROP_JSON_READER_POOL = 1,
+  PROP_JSON_READER_USE_THREADS,
+  PROP_JSON_READER_BLOCK_SIZE,
+  PROP_JSON_READER_ALLOW_NEWLINES_IN_VALUES,
+  PROP_JSON_READER_UNEXPECTED_FIELD_BEHAVIOR
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE(GArrowJSONReadOptions,
+                           garrow_json_read_options,
+                           G_TYPE_OBJECT)
+
+#define GARROW_JSON_READ_OPTIONS_GET_PRIVATE(object) \
+  static_cast<GArrowJSONReadOptionsPrivate *>(       \
+    garrow_json_read_options_get_instance_private(   \
+      GARROW_JSON_READ_OPTIONS(object)))
+
+static void
+garrow_json_read_options_set_property(GObject *object,
+                                      guint prop_id,
+                                      const GValue *value,
+                                      GParamSpec *pspec)
+{
+  auto priv = GARROW_JSON_READ_OPTIONS_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_JSON_READER_POOL:
+    priv->pool = static_cast<arrow::MemoryPool *>(g_value_get_pointer(value));
+    break;
+  case PROP_JSON_READER_USE_THREADS:
+    priv->read_options.use_threads = g_value_get_boolean(value);
+    break;
+  case PROP_JSON_READER_BLOCK_SIZE:
+    priv->read_options.block_size = g_value_get_int(value);
+    break;
+  case PROP_JSON_READER_ALLOW_NEWLINES_IN_VALUES:
+    priv->parse_options.newlines_in_values = g_value_get_boolean(value);
+    break;
+  case PROP_JSON_READER_UNEXPECTED_FIELD_BEHAVIOR:
+    priv->parse_options.unexpected_field_behavior =
+      static_cast<arrow::json::UnexpectedFieldBehavior>(g_value_get_enum(value));
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+garrow_json_read_options_get_property(GObject *object,
+                                      guint prop_id,
+                                      GValue *value,
+                                      GParamSpec *pspec)
+{
+  auto priv = GARROW_JSON_READ_OPTIONS_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_JSON_READER_USE_THREADS:
+    g_value_set_boolean(value, priv->read_options.use_threads);
+    break;
+  case PROP_JSON_READER_BLOCK_SIZE:
+    g_value_set_int(value, priv->read_options.block_size);
+    break;
+  case PROP_JSON_READER_ALLOW_NEWLINES_IN_VALUES:
+    g_value_set_boolean(value, priv->parse_options.newlines_in_values);
+    break;
+  case PROP_JSON_READER_UNEXPECTED_FIELD_BEHAVIOR:
+    g_value_set_enum(value, static_cast<char>(priv->parse_options.unexpected_field_behavior));
```

Comment:
Could you use `gint` instead of `char`?
Because `g_value_set_enum()` uses `gint`: https://developer.gnome.org/gobject/stable/gobject-Standard-Parameter-and-Value-Types.html#g-value-set-enum

Reply:
I've fixed it.